vault/data_source_ldap_static_role_credentials.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/vault/api"
+)
+
+func ldapStaticCredDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadContextWrapper(readLDAPStaticCreds),
+		Schema: map[string]*schema.Schema{
+			consts.FieldPath: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "LDAP Secret Backend to read credentials from.",
+			},
+			consts.FieldRoleName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role.",
+			},
+			consts.FieldDN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Distinguished name (DN) of the existing LDAP entry to manage password rotation for.",
+			},
+			consts.FieldLastVaultRotation: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last time Vault rotated this static role's password.",
+			},
+			consts.FieldPassword: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Password for the static role.",
+				Sensitive:   true,
+			},
+			consts.FieldLastPassword: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last known password for the static role.",
+				Sensitive:   true,
+			},
+			consts.FieldRotationPeriod: {
+				Type:        schema.TypeString,
```

Comment:
I think in the past we learned that it is easier to manage Vault duration strings and periods as `TypeInt` in the TFVP. This is because in Vault the duration string that we input is not necessarily the string that you get back, and this causes a problem for the TFVP state. For example, if in the test below, we update the static role `rotation_period` to be `"1m"` (which is a valid input string for the current schema), Vault will perform some operations on it, and store and return the value `"60"` back to TFVP on a read. This causes a drift in the TF state since `"1m" != "60"`, and this gets hard to mitigate unless users know exactly how to enter duration strings so Vault does not manipulate the return values. If you run the test with that change, the test will fail with a TF state drift.

Using `TypeInt` avoids all these weird cases; whatever value the user wants to input is interpreted as seconds, and written to Vault as a string; for example, if the input was 7200 for the field, we write `fmt.Sprintf("%ds", duration)`, which would write the field as `"7200s"` to Vault. Now on read there should be no drift, and we can get the integer value from the returned string and set it to state. 

With this context, looking back at the static role resource as well as this resource, I think it would make our lives easier in the future and we would avoid some potential bug cases if we handle all durations and periods as integers. There are some examples in the repo that shouldn't be too hard to find, but let me know if you have any issues and I'm happy to talk more about this! Hopefully all of that made sense, apologies for the long comment 🙏🏼

Reply:
Ah, good to know! Updated~

Reply:
Nice! Should we also update the `rotation_period` on the static role resource in this PR or should we take care of that in a later PR? Either is fine by me as long as we remember to make the change :)

Reply:
should be handled [here](https://github.com/hashicorp/terraform-provider-vault/pull/1821/files#diff-106def9fd3cdaf4291ab380ac4a4ddf92bd44d4fe3741a934e88f1e20baa124dR45)!